### PR TITLE
Bug fix/6215 export to qc failed tobe exported

### PR DESCRIPTION
**Pull request by uzanh**

### CODE REVIEW

File: Ginger/Ginger/ALM/ALMIntegration.cs [hunk 1]
```diff
@@ -489,48 +489,86 @@ public ObservableList<ExternalItemFieldBase> GetALMItemFieldsREST(bool online, A
...
-                //Sorting the list
-                ObservableList<ExternalItemFieldBase> sortedFields = new ObservableList<ExternalItemFieldBase>(from i in exitingFields orderby i.ItemType select i);
                 exitingFields.ClearAll();
-                exitingFields.Append(sortedFields);
+                exitingFields.Append(mergedFields);
+            }
+
+        }
+        internal ObservableList<ExternalItemFieldBase> getUpdatedFields(ObservableList<ExternalItemFieldBase> mItemsFields, bool online, BackgroundWorker bw = null)
+        {
```
Comment (MeniKadosh1) [inline Ginger/Ginger/ALM/ALMIntegration.cs]: Function name getUpdatedFields should start with Capital letter like : "GetUpdatedFields "

File: Ginger/Ginger/ALM/ALMItemsFieldsConfigurationPage.xaml.cs [hunk 2]
```diff
@@ -88,12 +88,15 @@ public void ShowAsWindow(eWindowShowStyle windowStyle = eWindowShowStyle.Dialog)
...
             ObservableList<ExternalItemFieldBase> tempItemList = new ObservableList<ExternalItemFieldBase>();
-            foreach (ExternalItemFieldBase field in mItemsFields.Where(x => x.ToUpdate == true).ToList())
-                tempItemList.Add(field);
-            App.UserProfile.Solution.ExternalItemsFields = tempItemList;
+            //foreach (ExternalItemFieldBase field in mItemsFields.Where(x => x.ToUpdate == true).ToList())
+            //    tempItemList.Add(field);
+            App.UserProfile.Solution.ExternalItemsFields = ALMIntegration.Instance.getUpdatedFields(mItemsFields, false);
             App.UserProfile.Solution.SaveSolution(true, SolutionGeneral.Solution.eSolutionItemToSave.ALMSettings);
+            App.UserProfile.Solution.ExternalItemsFields = mItemsFields;
             genWin.Close();
```
Comment (MeniKadosh1) [inline Ginger/Ginger/ALM/ALMItemsFieldsConfigurationPage.xaml.cs]: why need to set Solution.ExternalItemsFields = mItemsFields after save? it loses it in save process?

File: Ginger/GingerCore/ALM/ALMCore.cs [hunk 3]
```diff
@@ -80,5 +80,6 @@ public BusinessFlow ConvertRallyTestPlanToBF(RallyTestPlan testPlan)
         {
             get; set;
         }
+        public abstract ObservableList<ExternalItemFieldBase> almItemFields { get; set; }
     }
```
Comment (MeniKadosh1) [inline Ginger/GingerCore/ALM/ALMCore.cs]: Public field need to be with capital first letter like "AlmItemFields"

File: Ginger/GingerCore/ALM/QCCore.cs [hunk 4]
```diff
@@ -80,7 +80,13 @@ public override ObservableList<Activity> GingerActivitiesRepo
 
         public override ObservableList<ExternalItemFieldBase> GetALMItemFields(BackgroundWorker bw, bool online, ALM_Common.DataContracts.ResourceType resourceType)
         {
-            return ImportFromQC.GetALMItemFields();
+            ObservableList<ExternalItemFieldBase> tempFieldsList = ImportFromQC.GetALMItemFields();
+            almItemFields = new ObservableList<ExternalItemFieldBase>();
+            foreach (ExternalItemFieldBase item in tempFieldsList)
+            {
+                almItemFields.Add((ExternalItemFieldBase)item.CreateCopy());
+            }
```
Comment (MeniKadosh1) [inline Ginger/GingerCore/ALM/QCCore.cs]: seems like GetALMItemFields function is same for all QC/RQM/Rally no? can't we place it in ALMCore instead having it in each alm type core class?

File: Ginger/Ginger/ALM/ALMItemsFieldsConfigurationPage.xaml.cs (hunk 2 above)
Comment (uzanh) [reply Ginger/Ginger/ALM/ALMItemsFieldsConfigurationPage.xaml.cs]: sending for save only the "changed from default" and after save set all fields to cache, i will change it while jira development, for now i just improve the old code.

File: Ginger/GingerCore/ALM/QCCore.cs [hunk 5]
```diff
@@ -81,7 +81,13 @@ public override ObservableList<Activity> GingerActivitiesRepo
 
         public override ObservableList<ExternalItemFieldBase> GetALMItemFields(BackgroundWorker bw, bool online, ALM_Common.DataContracts.ResourceType resourceType)
         {
-            return ImportFromQC.GetALMItemFields();
+            ObservableList<ExternalItemFieldBase> tempFieldsList = ImportFromQC.GetALMItemFields();
+            AlmItemFields = new ObservableList<ExternalItemFieldBase>();
```
Comment (MeniKadosh1) [inline Ginger/GingerCore/ALM/QCCore.cs]: this 5 lines of code are the same for all GetALMItemFields in all ALM types, can we do it in some generic function? can be some function on the base class ALMCore

**Comment by MeniKadosh1** [comment]
merged and deleted brance

**Pull request merged**